### DISCUSSION TITLE
Magic constants in declaration files?

### DISCUSSION THREAD

**Discussion by DeidaraMC** [teal-language/tl#808] [Q&A] [answered]
How can we add magic constants to declaration file functions? I.e. indicate that a function accepts a string but only expects a few different magic string values? 
With emmylua notation something like this can be done to achieve the desired effect: 
---@field MyFunction fun(self, service: "MagicString1"): OptionA
---@field MyFunction fun(self, service: "MagicString2"): OptionB

**Comment by lenscas** [comment, marked as answer]
You can use enums.

**Reply by DeidaraMC** [reply]
How about the functionality of returning a different option based on the enum provided?

**Reply by lenscas** [reply]
there is a little hack you can do using multiple enums and overloading. It is used here https://github.com/teal-language/teal-types/blob/master/types/luasql/postgres.d.tl to return different types based on the type given.

**Reply by DeidaraMC** [reply]
Thank you, is there a way to have the lsp help a bit more aggressively? Ideally the potential options should just appear with an intellisense box as I start to type allowing me to tab complete and see all definitions with ease. This is the behaviour seen with emmylua notation in general if there's a way to make the lsp hinting a bit more aggressive that would be really nice/user friendly.
Seems like even a single enum with multiple options provides no help as you type into a function which accepts them.
